File: WebSite/src/app/tools/request/Pager.ts
export interface Page {
  PageSize: number;
  PageIndex: number;
}

export type Pager<T extends any = {}> =
  Page
  & {
    Rows?: T[];
    ShowingFirstRowIndex: number;
    ShowingLastRowIndex: number;
    TotalPageCount: number;
    TotalRecord: number;
    ViewingRecord: number;
  };

export const getEmptyPager = <T>(): Pager<T> => ({
  PageSize: 25,
  PageIndex: 0,
  Rows: [],
  ShowingFirstRowIndex: 0,
  ShowingLastRowIndex: 0,
  TotalPageCount: 1,
  TotalRecord: 0,
  ViewingRecord: 0
});

File: API/Entities/Pager.cs
using System.Collections.Generic;

namespace Entities
{
    public class Pager<T>
    {
        public long PageSize { get; set; }
        public long PageIndex { get; set; }
        public List<T> Rows { get; set; }
        public long ShowingFirstRowIndex { get; set; }
        public long ShowingLastRowIndex { get; set; }
        public long TotalPageCount { get; set; }
        public long TotalRecord { get; set; }
        public long ViewingRecord { get; set; }
    }
}
